src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -680,48 +804,62 @@ BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, IN_MEMORY_READ_SEQUENTIAL)(b
 BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, IN_MEMORY_READ_RANDOM)(benchmark::State& state) {
   for (auto _ : state) {
     state.PauseTiming();
-    const auto random_indices = generate_random_indexes(NUMBER_OF_ELEMENTS);
+    create_random_indexes_if_needed(state.range(0), NUMBER_OF_ELEMENTS);
     auto read_data = std::vector<uint32_t>{};
     read_data.resize(NUMBER_OF_ELEMENTS);
     state.ResumeTiming();
 
-    for (auto index = size_t{0}; index < NUMBER_OF_ELEMENTS; ++index) {
-      read_data[index] = numbers[random_indices[index]];
+    for (auto index = uint64_t{0}; index < NUMBER_OF_ELEMENTS; ++index) {
+      read_data[index] = numbers[random_indexes[index]];
     }
 
     state.PauseTiming();
     const auto sum = std::accumulate(read_data.begin(), read_data.end(), uint64_t{0});
 
-    Assert(control_sum == static_cast<uint64_t>(sum), "Sanity check failed: Not the same result");
-    Assert(&read_data[0] != &numbers[random_indices[0]], "Sanity check failed: Same reference");
+    Assert(control_sum == sum, "Sanity check failed: Not the same result. Got: " + std::to_string(sum) +
+                                   " Expected: " + std::to_string(control_sum) + ".");
+    Assert(&read_data[0] != &numbers[random_indexes[0]], "Sanity check failed: Same reference");
 
     state.ResumeTiming();
   }
 }
 
 // Arguments are file size in MB
+
+static void CustomArguments(benchmark::internal::Benchmark* benchmark) {
+  const std::vector<uint32_t> parameters = {10000, 100000};
+  const std::vector<uint8_t> thread_counts = {1, 2, 4, 8, 16, 24, 32, 40, 48, 56, 64};
+
+  for (auto param_index = size_t{0}; param_index < parameters.size(); ++param_index)
+    for (auto thread_index = size_t{0}; thread_index < thread_counts.size(); ++thread_index)
+      benchmark->Args({parameters[param_index], thread_counts[thread_index]});
+}
+
+
```

Comment:
If you insist on using an own dedicated function for this, some thoughts:
1. Rename `CustomArguments` -> `custom_arguments` (Hyrise Style Guidelines)
2. Rename `parameters` to `sizes`.
3. Find a way to make this a three-liner to be a good never-nester. Line 1: Sizes, Line 2: Thread Counts, Line 3: Return both.
4. Keep in mind, that this should reduce LOC, but currently it adds 10 LOC without new functionality.

I imagine something like this:
```cpp
const std::vector<uint32_t> sizes= {10000, 100000};
const std::vector<uint8_t> thread_counts = {1, 2, 4, 8, 16, 24, 32, 40, 48, 56, 64};
return {sizes, thread_counts}
```